content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/architecture.md

```diff
@@ -0,0 +1,143 @@
+
+## Component design
+
+The specific functional and data separations in the terrain component design have been chosen to make it easy to control what functionality is active and to provide clear touchpoints for replacing functionality everywhere that it's likely to be useful.
+
+### Level components
+
+The terrain system uses three level components:
+
+| Component | Description |
+| - | - |
+| **Terrain World** | Controls the base terrain system. The entire system is enabled or disabled by adding or removing this component. The configuration parameters that affect the entire terrain world exist on this component as well.|
+| **Terrain World Renderer** | Controls the terrain rendering. It requires the **Terrain World** component to exist, and it provides all of the rendering configuration that needs to be consistent across the entire terrain. This component is kept separate from **Terrain World** so that it's possible to define a conceptual terrain without requiring the overhead of rendering for use cases where a visual representation simply isn't necessary. Also, this provides a "replacement touchpoint" for creating different terrain renderers, such as a voxel-based implementation or a raytracing-based implementation.|
+| **Terrain World Debugger** | Provides helpful debugging features for visualizing aspects of the terrain system. This component is kept separate from **Terrain World** so that it can easily be completely removed from shipping products.|
+
+### Entity components
+
+There are two types of terrain entity components - the "core" components that are used to define a terrain region and "auxiliary" components that operate on an area that can contain terrain regions.
+
+Core Components:
+
+| Component | Description |
+| - | - |
+| **Axis-Aligned Box Shape** | Defines the shape of a terrain region, including the min/max heights. |
+| **Terrain Layer Spawner** | Declares that a terrain region exists in the shape bounds, and it has a specified priority. |
+| **Terrain Height Gradient List** | (Optional) Defines the height data for the terrain region. |
+| **Terrain Surface Gradient List** | (Optional) Defines the surface type and weight data for the terrain region. |
+
+Auxiliary Components:
+
+The "auxiliary" components only require a Box Shape on their entity, but not a Terrain Layer Spawner, so they can span across multiple terrain regions instead of being tied to a single one. They _can_ still appear on an entity with a Terrain Layer Spawner, it just isn't required.
+
+| Component | Description |
+| - | - |
+| **Terrain Macro Material** | Defines terrain base coloring for all terrain regions that fall within its volume. Macro materials aren't directly attached to single terrain regions so that they can be authored and streamed at different world resolutions than the terrain regions. |
```

Comment:
```suggestion
| **Terrain Macro Material** | Defines terrain base color for all terrain regions that fall within its volume. Macro materials aren't directly attached to single terrain regions so that they can be authored and streamed at different world resolutions than the terrain regions. |
```

Reply:
I went with "coloring" instead of "color" because "color" sounds a little too singular, and "colors" sounds more like a list of colors than an applied texture. I'm open to more suggestions here though. Maybe "Applies a terrain base color texture to all terrain regions that fall within its volume."?

Here are some examples of what the macro material does to the terrain:
![level-with-macro-color](https://user-images.githubusercontent.com/82224783/204674966-c935a3b1-d4a5-439c-87df-19f5ae8821b7.png)
![level-with-macro-color-up-close](https://user-images.githubusercontent.com/82224783/204675002-ed9e1ff5-1a06-4517-8808-24e36cae8585.png)

Reply:
Yes I think "a terrain base color texture" is perfect. Texture is a clue to the reader what kind of asset is used.